python/python/lance/lance/indices/__init__.pyi

```diff
@@ -47,3 +50,24 @@ def transform_vectors(
     pq_codebook: pa.Array,
     dst_uri: str,
 ): ...
+
+class IndexSegmentDescription:
+    uuid: str
+    dataset_version: int
```

Comment:
Do we want a better name that this? As-is, I think it's unclear which version this is supposed to be.